packages/common-utils/src/renderChartConfig.ts

```diff
@@ -499,39 +501,70 @@ async function timeFilterExpr({
   timestampValueExpression: string;
   with?: ChartConfigWithDateRange['with'];
 }) {
-  const valueExpressions = splitAndTrimWithBracket(timestampValueExpression);
   const startTime = dateRange[0].getTime();
   const endTime = dateRange[1].getTime();
 
+  let optimizedTimestampValueExpression = timestampValueExpression;
+  try {
+    // Not all of these will be available when selecting from a CTE
+    if (databaseName && tableName && connectionId) {
+      const { primary_key } = await metadata.getTableMetadata({
+        databaseName,
+        tableName,
+        connectionId,
+      });
+      optimizedTimestampValueExpression = optimizeTimestampValueExpression(
+        timestampValueExpression,
+        primary_key,
+      );
+    }
+  } catch (e) {
+    console.log('Failed to optimize timestampValueExpression', e);
+  }
```

Comment:
Clicking through the app, I don't run into this error, and don't expect it to happen, but am adding this try catch here as an extra layer of protection against an error either in getTableMetadata or optimizeTimestampValueExpression, in which case we can always fall back to the raw `timestampValueExpression` used before